sqlglot/parser.py

```diff
@@ -3259,21 +3263,34 @@ def _parse_set_operations(self, this: t.Optional[exp.Expression]) -> t.Optional[
         token_type = self._prev.token_type
 
         if token_type == TokenType.UNION:
-            expression = exp.Union
+            operation = exp.Union
         elif token_type == TokenType.EXCEPT:
-            expression = exp.Except
+            operation = exp.Except
         else:
-            expression = exp.Intersect
+            operation = exp.Intersect
+
+        comments = self._prev.comments
+        distinct = self._match(TokenType.DISTINCT) or not self._match(TokenType.ALL)
+        by_name = self._match_text_seq("BY", "NAME")
+        expression = self._parse_set_operations(
+            self._parse_select(nested=True, parse_set_operation=False)
+        )
+
+        set_operation_modifiers: t.Dict[str, t.Any] = {}
+        if self.MODIFIERS_ATTACHED_TO_UNION and isinstance(expression, exp.Expression):
+            for arg in self.UNION_MODIFIERS:
```

Comment:
can we not do a for loop?

Reply:
will be a bit verbose / duplicative because we have to check if the args exist etc

Reply:
so the for loop is just to compact that a bit